### PR TITLE
Fix String initialiser type when java.lang.String is a stub

### PR DESCRIPTION
**Pull request by owen-mc-diffblue**
Previously it was accidentally given type java.lang.Object, but we
didn't notice because we were less stringent about verifying type
equality (or, in transitioning from ID_symbol to ID_struct_tag for
Java types, we've become less generous about permitting a structural
match but name mismatch). Either way it was clearly wrong to assign
a wrong type like this.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [x] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [x] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Comment by smowton** [comment]
@owen-jones-diffblue please add a test for this -- `janalyzer` is actually a perfect vessel to test this as it can run the constant propagator, which checks for type consistency. I can reproduce the crash with `janalyzer test.class --location-sensitive --constants --show`, where `test.class` contains any string literal operations.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 1365f8b94e0d08cc49d53eace0f7e968b4c64e27).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/96094506

**Comment by owen-mc-diffblue** [comment]
I've added a test using janalyzer, and also a fix required to get janalyzer to run properly.

**Review by smowton — APPROVED** [review]

File: jbmc/regression/CMakeLists.txt [hunk 1]
```diff
@@ -7,6 +7,7 @@ add_subdirectory(jbmc)
 add_subdirectory(strings-smoke-tests)
 add_subdirectory(jbmc-strings)
 add_subdirectory(jdiff)
+add_subdirectory(janalyzer)
```
Comment (tautschnig) [inline jbmc/regression/CMakeLists.txt:10]: Please also add it to the `Makefile`.

Comment (owen-mc-diffblue) [reply jbmc/regression/CMakeLists.txt:10]: Oops. Well spotted.

**Review by allredj — COMMENTED** [review]
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: 101e46eb4e0ab33af5698694e450e7bf14460948).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/96178128
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 0d96b0d08fece7d8e539e063010c428d252d4f9a).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/96192263

**Review by NathanJPhillips — APPROVED** [review]

**Pull request merged**